### ISSUE TITLE
Switch to JSON & application/json content type in all requests to GitLab API

### ISSUE THREAD

**Issue by gdubicki** [gitlabform/gitlabform#90]
As it turned out in https://gitlab.com/gitlab-org/gitlab/-/issues/202216#note_302059234 we should have been using JSON & `application/json` content type in PUT/POST request to the GitLab API to avoid issues like #68, #69 and #70.

I would try to switch to it everywhere, completely stopping to use `data` parameter of `requests` and using `json` everywhere.

This way there will be no examples of code that may lead to issues later on (as ALMOST EVERYTHING does work when you use URL encoded form & content type set to `application/x-www-form-urlencoded` - only at some point it stops...) and the chance that we will have this kind of problems in the new features is minimalized.

However to do this change properly we should add all the missing tests to ensure that the app won't stop working.

PS If you want to help to develop this project then this is a good starting point! Please let us know.

**Comment by jimisola** [comment]
Is this one relevant once all API calls have been migrated to python-gitlab as per #73?

**Comment by amimas** [comment]
Yeah, this is probably not relevant once everything has been migrated over to python-gitlab. In fact, I suggest we remove the `help wanted` and `good for first issue` labels. It'd be waste of time to spend any time/effort in this issue.

**Comment by amimas** [comment]
I'm going to close this issue to help reduce number of open issues. There is no plan to work on this as mentioned above since our goal is to migrate to python-gitlab library instead of adding/maintaining API to this repo.
